### DISCUSSION TITLE
Auto generate column names for composite columns

### DISCUSSION THREAD

**Discussion by klinden** [sqlalchemy/sqlalchemy#10627] [Ideas]
Composite columns are already able to auto generate data types from the base Python dataclass. It would be nice if they could automatically generate a name as well. I realize it's probably impossible to standardize on something everybody would love, but it would be nice to have a reasonable default convention. For example:

```python
@dataclass
class NumericProperty:
    min: float
    max: float
    value: float
```

```python
class Thing(Base):
    weight: Mapped[NumericProperty] = composite()
```

In this case, three float columns named `weight_min`, `weight_max`, and `weight_value` could be generated.

**Comment by zzzeek** [comment]
so we really avoid at all costs ever hardcoding DDL naming conventions inside of SQLAlchemy, because they lead to confusion as to where things are happening and give the impression that SQLAlchemy might be making implicit decisions sometimes, which also adds to the guessing one must do when things aren't working.   There is of course a single naming convention stuck inside of "Index" which we are kind of stuck with, but otherwise there should be none.

**Reply by klinden** [reply]
Ok, thanks for the reply.
